tmc-langs-util/src/test/java/fi/helsinki/cs/tmc/langs/util/MainTest.java

```diff
@@ -0,0 +1,203 @@
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testRunTests() {
+        String exercisePath = getTargetPath("arith_funcs");
+        String outputPath = exercisePath + "/results.txt";
+        String[] args = {"--runtests", exercisePath, outputPath};
+        expectedMessage = "Test results can be found in " + outputPath;
+        mainTest(0, args);
+    }
+
+    @Test
+    public void testRunCheckCodeStyle() {
+        String exercisePath = getTargetPath("arith_funcs");
+        String outputPath = exercisePath + "/exercises.txt";
+        String[] args = {"--checkstyle", exercisePath, outputPath};
+        expectedMessage = "Codestyle report can be found at " + outputPath;
+        mainTest(0, args);
+    }
+
+    @Test
+    public void testPrepareStub() {
+        String[] args = {"--preparestub", getTargetPath("arith_funcs")};
+
+        final TaskExecutor executor = Mockito.mock(TaskExecutor.class);
+        final Path stubPath = new File(getTargetPath("arith_funcs")).toPath();
+
+        Main mainClass = new Main(executor);
+        exit.expectSystemExitWithStatus(0);
+        exit.checkAssertionAfterwards(new Assertion() {
+            @Override
+            public void checkAssertion() throws Exception {
+                Mockito.verify(executor).prepareStub(stubPath);
+            }
+        });
+        mainClass.main(args);
+    }
+
+    @Test
+    public void testPrepareSolution() {
+        String[] args = {"--preparesolution", getTargetPath("arith_funcs")};
+
+        final TaskExecutor executor = Mockito.mock(TaskExecutor.class);
+        final Path solutionPath = new File(getTargetPath("arith_funcs")).toPath();
+
+        Main mainClass = new Main(executor);
+        exit.expectSystemExitWithStatus(0);
+        exit.checkAssertionAfterwards(new Assertion() {
+            @Override
+            public void checkAssertion() throws Exception {
+                Mockito.verify(executor).prepareSolution(solutionPath);
+            }
+        });
+        mainClass.main(args);
+    }
+
+    /**
+     * Call main method with given args, and assert that main exits with given
+     * exit status. Additionally do assertion that main prints out
+     * {@link #expectedMessage}.
+     *
+     * @param exitStatus           expected exit status for main.
+     * @param args                 for calling main.
+     * @param optionalErrorMessage for system out assertion.
+     */
+    private void mainTest(int exitStatus, String[] args, String... optionalErrorMessage) {
+        exit.expectSystemExitWithStatus(exitStatus);
+        exitStringContainsAssertion(expectedMessage, optionalErrorMessage);
+        Main.main(args);
+    }
+
+    private void mainTestEquals(int exitStatus, String[] args) {
+        exit.expectSystemExitWithStatus(exitStatus);
+        exitStringEqualsAssertion(expectedMessage);
+        Main.main(args);
+    }
```

Comment:
I think I'd remove these two methods - could just do the setup inline in the tests like you already do in `testPrepareStub` and `testPrepareSolution`. Alternatively try to name these better.

Not a very strong opinion. DRY vs clarity in tests is often debatable.